### PR TITLE
Tagfilter: Adding no result message feature

### PR DESCRIPTION
**Pull request by Garneauma**
Adding the possibility to define a "No result" message element that will show when there are no items visible.

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]
(quick partial review)

* [x] Rename `sr-only` by `wb-inv`

File: src/plugins/tagfilter/tagfilter.js
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter.js:59]: Not sure if the only use of the `[aria-live=polite]` is enough to satisfy at [SC 4.1.3 Status messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html).

I do think the "Situation A" is applicable here because the status message advises on the results of an action.

According to the sufficient technique, it might require to use the role "status".

Also, to be tested with 1 AT, is the show/hide of the content block do get announced to AT user when:
1 - There is a few result
2 - There is no result (the block is displayed for the first time)
3 - There is a few result
4 - There is no result ( the block is re-displayed for a second time)

What about then the component do have 2 status content block? like the "Example 3". I do think the content block representing the status "no results" should be marked with the role "status" also.

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/tagfilter/tagfilter.js
Comment (Garneauma) [reply src/plugins/tagfilter/tagfilter.js:59]: When I tested using the Mac screen-reader, I correctly got the message for no results every time (hiding, then showing, then hiding, etc.). However, I can add the role status to make it more obvious for the screen reader, and test.

Comment (duboisp) [reply src/plugins/tagfilter/tagfilter.js:59]: ok it is compatible with AT on mac. Have you compared against the SC 4.1.3 statement? Is it satisfied and applicable technique applied?

Comment (Garneauma) [reply src/plugins/tagfilter/tagfilter.js]: Switched `aria-live="polite"` to `role="status"` which satisfies success criterion 4.1.3. However, while fixing this, I have discovered an issue with the current implementation which does not helpfully declare de changes in the DOM when a filter is selected. This issue is going to be fixed in an upcoming PR.

**Comment by duboisp** [comment]
Pre-approved upon second review and testing.

**Review by duboisp — CHANGES_REQUESTED** [review]
Le "no result" message n'est pas fonctionnel. Je pense qu'il manque ce bloc de code avec le mutation observer:

https://github.com/wet-boew/wet-boew/commit/e061bdcce03adbb4616657943bce94301a918969#diff-40069d6c91d0286504cff47f4416f809a249c4f1b151a99974e78fc682c34880R61-R82

File: src/plugins/tagfilter/tagfilter.js
Comment (Garneauma) [reply src/plugins/tagfilter/tagfilter.js:1]: @duboisp From our discussion, the feature does not work on Firefox due to the use of the CSS selector `:has()`. A fallback solution will need to be implemented for Firefox.

**Comment by duboisp** [comment]
Pre-approved upon successful second review and local testing.

File: src/plugins/tagfilter/tagfilter-fr.hbs
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter-fr.hbs:1]: The no result don't appear in the French example. But it do work for the English example.

File: src/plugins/tagfilter/tagfilter.js
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter.js:227]: It will be preferable to do a [feature detection](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection) instead of being browser specific. Being being browser specific could let to unexpected behaviour when the browser would start to support the feature, especially when we know that some site take a lot of time before to update their WET-BOEW implementation to the latest version.

Can you check if it would be possible to use a polyfill?
I didn't tested but like: https://github.com/jplhomer/polyfill-css-has/blob/master/src/polyfill-css-has.ts
Can we leverage somehow the [@supports CSS at-rule](https://developer.mozilla.org/en-US/docs/Web/CSS/@supports) for this? Like if supported we style a little something that we can detect in JS to use an alternate implementation?

**Review by duboisp — CHANGES_REQUESTED** [review]
I did a review and local testing, here my observation:

* [x] French example don't work
* [x] Example 3 in English don't work in Firefox
* [x] Use feature detection instead of being browser specific

**Comment by duboisp** [comment]
Pre-approved once the change is pushed + review and local testing completed.

File: src/plugins/tagfilter/tagfilter.js
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter.js:276]: Considering the closure scope currently defined for this plugin, this call should be made from the `window` variable instead of calling the global directly.

```suggestion
	let supportsHas = window.getComputedStyle( document.documentElement ).getPropertyValue( "--supports-has" );
```

**Review by duboisp — CHANGES_REQUESTED** [review]
Awesome way to detect the `:has` CSS selector, I love it. 

I just have a little typographic change which are normal caught, I thought,  by the "use strict" instruction or our eslint task but I don't know why it didn't.

Todo:
* [x] Call `getComputedStyle` via the `window` scoped variable
* [x] Fix the merge conflict

Note: Once this PR is merged, you will need to replicate in GCWeb and adapt the change made at the build script. That sync is required in order to release GCWeb along with this changes.

File: src/plugins/tagfilter/tagfilter-fr.hbs
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter-fr.hbs:803]: I think this was added by mistake because it is already there at the end of the list of articles.

File: src/plugins/tagfilter/tagfilter.js
Comment (duboisp) [inline src/plugins/tagfilter/tagfilter.js:268]: There is an extra space

```suggestion
$document.on( "wb-contentupdated", selector, function() {
```

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — APPROVED** [review]
Reviewed and tested locally, work as expected in Chrome and in FF.

**Pull request merged**
